File: Java/Ciekawy wzorzec.java
import java.util.Scanner;

public class Main 
{
    public static boolean checkForPattern(String word, String pattern) 
    {
        int indexInPattern = 1;
        for (int i = 2; i < pattern.length() * 2; i += 2) 
        {
            if (word.charAt(i) == pattern.charAt(indexInPattern)) 
                indexInPattern++;
            else 
                return false;
        }
        return true;
    }

    public static void main(String[] args) 
    {
        int index = -1;
        Scanner input = new Scanner(System.in);
        String word = input.nextLine();
        String pattern = input.nextLine();
        for (int i = 0; i < word.length(); i++) 
            if (word.charAt(i) == pattern.charAt(0))
                if (checkForPattern(word.substring(i), pattern)) 
                {
                    index = i + 1;
                    break;
                }
        System.out.print(index == -1 ? "brak" : index);
        input.close();
    }
}

File: Python/Ciekawy wzorzec.py
def check_for_pattern(word, pattern):
    index_in_pattern = 1
    for i in range(2, len(pattern) * 2, 2):
        if word[i] == pattern[index_in_pattern]:
            index_in_pattern += 1
        else:
            return False
    return True

index = -1
word = input()
pattern = input()
for i in range(len(word)):
    if word[i] == pattern[0]:
        if check_for_pattern(word[i:], pattern):
            index = i + 1 
            break
print('brak' if index == -1 else index)